File: scratch/query_db.dart
import 'dart:io';
import 'package:sqflite_common_ffi/sqflite_ffi.dart';
import 'package:path/path.dart' as p;

void main() async {
  // Initialize sqflite ffi
  sqfliteFfiInit();
  var databaseFactory = databaseFactoryFfi;
  
  final docDir = 'C:\\Users\\ASUS\\Documents';
  final dbPath = p.join(docDir, 'nature_biotic_local.db');
  
  print("Opening DB at: $dbPath");
  if (!File(dbPath).existsSync()) {
    print("Database file does not exist.");
    return;
  }
  
  var db = await databaseFactory.openDatabase(dbPath);
  
  try {
    var rows = await db.rawQuery("SELECT * FROM store_transactions ORDER BY created_at DESC LIMIT 40;");
    print("\nStore Transactions (Last 40):");
    for (var r in rows) {
      print("${r['id']} | type: ${r['transaction_type']} | item: ${r['item_name']} | qty: ${r['quantity']} | unit: ${r['unit']} | status: ${r['status']} | date: ${r['created_at']}");
    }
  } catch (e) {
    print("Error querying store_transactions: $e");
  }
  
  await db.close();
}

File: scratch/query_db.py
import sqlite3
import os

db_path = os.path.expanduser(r'~\Documents\nature_biotic_local.db')
print("DB Path:", db_path)

if not os.path.exists(db_path):
    print("Database file does not exist at path.")
else:
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    # List tables
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
    tables = cursor.fetchall()
    print("Tables in DB:", tables)
    
    # Query store_transactions
    try:
        cursor.execute("SELECT * FROM store_transactions ORDER BY created_at DESC LIMIT 20;")
        columns = [d[0] for d in cursor.description]
        rows = cursor.fetchall()
        print("\nStore Transactions (Last 20):")
        print("-" * 100)
        print(" | ".join(columns))
        print("-" * 100)
        for r in rows:
            print(" | ".join(str(val) for val in r))
    except sqlite3.OperationalError as e:
        print("Error reading store_transactions:", e)
        
    conn.close()
